steenroder/__init__.py

```diff
@@ -1,19 +1,19 @@
 import numpy as np
 from itertools import combinations
-from numba import jit
+from numba import njit
 
 
-@jit(nopython=True)
+@njit
 def _pivot(column):
-    try:
-        return max(column.nonzero()[0])
-    except:
-        return -1
+    nonzero = column.nonzero()[0]
+    if len(nonzero):
+        return max(nonzero)
+    return -1
 
 
 def get_boundary(filtration):
     spx_filtration_idx = {tuple(v): idx for idx, v in enumerate(filtration)}
-    boundary = np.zeros((len(filtration), len(filtration)), dtype=np.bool)
+    boundary = np.zeros((len(filtration), len(filtration)), dtype=bool)
```

Comment:
New versions of numpy state that `dtype=np.bool` is deprecated and one should just pass the in-built Python `bool`.  Same for `np.float` etc.  Of course, when using `numba` neither `np.bool` nor `bool` is correct, see below.